### PR TITLE
Hide 'Assign to Section' button when course is unassignable

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR hides the 'Assign to sections' button on the unit overview page when a course is not `assignable`.
`assignable` is a property of a `course_offering`. 

<details>

<summary>Background context on `course_version`</summary>

`course_version` is the 'link' between a [`unit`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/unit.rb) or [`unit_group`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/unit_group.rb) and [`course_offering`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/course_offering.rb). 

Below is a simplified visualization of the relationships among these models:

Curriculum version models

</details>

I updated the definition of `course_assignable?` in `assignable_course.rb` to check that the `course_version.course_offering.assignable` is `true`. I then updated unit tests in `assignable_course_test.rb` and `unit_test.rb` to ensure each course had a `course_version`. Thanks to @bethanyaconnor for her help with this task!

BEFORE UPDATE:

Screen Shot 2023-07-05 at 12 42 24 PM   
   

AFTER UPDATE:

Screen Shot 2023-07-05 at 3 13 32 PM

## Links
[jira ticket](https://codedotorg.atlassian.net/browse/SL-938)
[Slack thread](https://codedotorg.slack.com/archives/C045UAX4WKH/p1688396596254339)

## Testing story

I tested locally to ensure both standalone units and courses with a collection of units (such as CS Principles) behave as expected, i.e., when the 'assignable' checkbox is unchecked, the 'Assign to sections' button is hidden.

AFTER UPDATE - video of CS in Algebra after 'assignable' checkbox is unchecked:

https://github.com/code-dot-org/code-dot-org/assets/107423305/d615b54d-b067-446f-8122-1ca249dd4ac1

AFTER UPDATE - video of CS Principles after 'assignable' checkbox is unchecked.

https://github.com/code-dot-org/code-dot-org/assets/107423305/f4aff9ae-dab7-4465-93d9-ba678be79888

## Follow-up work
When a section is already assigned to CS in Algebra, ‘Find a course’ is displayed instead of 'CS in Algebra' which is misleading. 

Example is the 2021-22 1st period section in screenshot below:

Find_a_course

[jira ticket](https://codedotorg.atlassian.net/browse/SL-955) for this.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/course_offering.rb [hunk 1]
```diff
@@ -214,6 +214,7 @@ def summarize_for_assignment_dropdown(user, locale_code)
         display_name: any_versions_launched? ? localized_display_name : localized_display_name + ' *',
         category: category,
         is_featured: is_featured?,
+        assignable: assignable?,
```
Comment (bethanyaconnor) [inline dashboard/app/models/course_offering.rb]: is this still necessary?

Comment (fisher-alice) [reply dashboard/app/models/course_offering.rb]: Was a remnant of a past approach - removed!

**Review by kobryan0619 — COMMENTED** [review]
This looks good to me, but admittedly I haven't worked with writing ruby tests before.  Based on what I see and researched, it looks good though!

**Review by kobryan0619 — APPROVED** [review]

**Comment by fisher-alice** [comment]
> This looks good to me, but admittedly I haven't worked with writing ruby tests before. Based on what I see and researched, it looks good though!

Thanks Kaitie! I hadn't either so I consulted with @bethanyaconnor before updating these.

**Pull request merged**